python/mscclpp_benchmark/allreduce_bench.py

```diff
@@ -147,12 +154,16 @@ def run_benchmark(
     if MPI.COMM_WORLD.size // N_GPUS_PER_NODE == 1:
         if memory.nbytes < 2**20:
             mscclpp_call = MscclppAllReduce2(mscclpp_group, memory, memory_out)
-        elif memory.nbytes < 2**29:
+        elif memory.nbytes < 2**21 if is_nvls_supported() else memory.nbytes < 2**29:
```

Comment:
This isn't a good way to handle this. We should be able to test all algorithms and pick the best. This looks very manual.

Reply:
I agree on this.

Reply:
updated, add `find_best_algo` function. Due to some algos are designed for small message size, exclude these algos from the candidates for large message size

Reply:
Looks great!